stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -195,3 +219,32 @@ function eigvals!(A::Hermitian{T,S}, B::Hermitian{T,S}; sortby::Union{Function,N
     return vals
 end
 eigvecs(A::HermOrSym) = eigvecs(eigen(A))
+
+# Note: No specilized LAPACK routines for Matrix+Symmetric and Symmetric+Matrix exist. Hence, the calls are forwarded to conventional Matrix eigvals!
+function eigvals(A::StridedMatrix{T}, B::Union{RealHermSymComplexHerm{T},Diagonal{T}}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    return eigvals!(eigencopy_oftype(A, T), Matrix{T}(B); sortby)
+end
+
+function eigvals(A::RealHermSymComplexHerm{T,<:StridedMatrix}, B::AbstractMatrix{T}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    return eigvals!(Matrix{T}(A), eigencopy_oftype(B, T); sortby)
+end
+
+function eigvals!(A::AbstractMatrix{T}, C::Cholesky{T, <:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    # Cholesky decomposition based eigenvalues
+    return eigvals!(UtiAUi!(A, C.U); sortby)
+end
+function eigvals(A::AbstractMatrix{T}, C::Cholesky{T, <:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    eigvals!(eigencopy_oftype(A, T), C; sortby)
+end
+
+# Bunch-Kaufmann (LDLT) based solution for generalized eigenvalues
+function eigvals(A::AbstractMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    eigvals!(copy(A), B; sortby)
+end
+function eigvals!(A::AbstractMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
```

Comment:
Since this method is not in-place (it doesn't change neither `A` nor `B`, should we remove it and move the code to `eigvals`? Then we would avoid having `copy(A)` in line 242, which is currently superfluous.

Reply:
[ACCEPTED] Modified several other functions too to remove all ! functions. Commit: https://github.com/aravindh-krishnamoorthy/julia/commit/034762443bbe3efcee0f38ed925b28f129669b1b